libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -506,23 +505,21 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         )
 
         ifNonNull(accessToken) { token ->
-            runInTelemetryContext { telemetry ->
-                logD(
-                    "MapboxMetricsReporter.init from MapboxNavigation main",
-                    telemetry.LOG_CATEGORY
-                )
-                MapboxMetricsReporter.init(
-                    navigationOptions.applicationContext,
-                    token,
-                    obtainUserAgent()
-                )
-                MapboxMetricsReporter.toggleLogging(navigationOptions.isDebugLoggingEnabled)
-                telemetry.initialize(
-                    this,
-                    navigationOptions,
-                    MapboxMetricsReporter,
-                )
-            }
+            logD(
+                "MapboxMetricsReporter.init from MapboxNavigation main",
+                MapboxNavigationTelemetry.LOG_CATEGORY
+            )
+            MapboxMetricsReporter.init(
+                navigationOptions.applicationContext,
+                token,
+                obtainUserAgent()
+            )
+            MapboxMetricsReporter.toggleLogging(navigationOptions.isDebugLoggingEnabled)
+            MapboxNavigationTelemetry.initialize(
```

Comment:
All the processing in `MapboxNavigationTelemetry` would now be running even if telemetry collection is disabled. Could we improve that control to avoid the unnecessary processing?

Reply:
that's a fair point. I see following options:

### 1. Keep as is
pros:
* telemetry events are sent whenever the SDK client turns off and on telemetry globally
* simple code

cons:
* `MapboxNavigationTelemetry` works but events aren't sent when telemetry is disabled 

### 2. Initiaize MapboxNavigationTelemetry only when telemetry is turned on, always disable on destroy 
pros:
* `MapboxNavigationTelemetry` doesn't work when telemetry is disabled globally 
* simple code

cons:
* telemetry events won't be sent if telemetry was turned off globally during initialization and then turned on in runtime

### 3. keep current telemetry state (enabled/disabled) on the sdk side, observe it and start/stop MapboxNavigationTelemetry
pros:
* telemetry events are sent whenever the SDK client turns off and on telemetry globally
* `MapboxNavigationTelemetry` doesn't work when telemetry is disabled globally 

cons:
* complex code 

## I proffer the option 1

It doesn't seem that `MapboxNavigationTelemetry` consumes a lot of resources while working, so I think option 1 is a reasonable tradeoff for us. @LukasPaczos , WDYT?

Reply:
I'd be in favor of option 2 that matches the existing behavior (avoid processing) and we already have NAVAND-355 that tracks the runtime enable/disable changes which we can tackle separately.

Reply:
Ok, let's keep current telemetry initialisation as is and improve it in NAVAND-355. Reverted cba98ab0d7c2d33e00f9b5ab3b2c285177628110